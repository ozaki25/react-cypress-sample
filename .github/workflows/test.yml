name: Test
on: [pull_request]
jobs:
  unit-test:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v1
      - name: Use Node.js 12
        uses: actions/setup-node@v1
        with:
          node-version: 12
      - name: install
        run: |
          yarn install
      - name: lint
        run: |
          yarn lint
      - name: test
        run: |
          yarn test
  e2e-test:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v1
      - name: Use Node.js 12
        uses: actions/setup-node@v1
        with:
          node-version: 12
      - name: install
        run: |
          yarn install
      - name: e2e-test
        run: |
          yarn e2e
