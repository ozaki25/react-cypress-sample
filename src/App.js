import React, { useState } from 'react';
import styled from 'styled-components';
import Title from './components/Title';
import Typography from './components/Typography';
import TextInput from './components/TextInput';
import Button from './components/Button';
import Container from './components/Container';
import { isUniformNumber } from './utils';
import { members } from './constants/members';

function App() {
  const [value, setValue] = useState('');
  const [valid, setValid] = useState(true);
  const [result, setResult] = useState(null);
  const onChange = e => setValue(e.target.value.trim());
  const onClick = () => {
    const isValid = isUniformNumber(value);
    setValid(isValid);
    setResult(null);
    if (isValid) {
      console.log({ members });
      setResult({ name: members[value], number: value });
      setValue('');
    }
  };
  return (
    <div>
      <Title>ファイターズ選手名鑑2019</Title>
      <Container>
        {!valid && <Typography error>背番号を入力して下さい</Typography>}
        <Typography>
          背番号が
          <TextInput onChange={onChange} value={value} />
          の選手は誰?
        </Typography>
        <Button onClick={onClick}>送信</Button>
        {result && (
          <Typography className="result">
            {`背番号 ${result.number} 番の選手は ${
              result.name ? `${result.name} です` : 'いません'
            }`}
          </Typography>
        )}
      </Container>
    </div>
  );
}

export default App;
