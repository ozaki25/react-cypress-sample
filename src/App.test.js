import React from 'react';
import { snapshot } from './test/helper';
import App from './App';

snapshot(<App />);
