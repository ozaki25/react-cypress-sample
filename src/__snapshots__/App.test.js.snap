// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot: App 1`] = `
.c0 {
  font-size: 3em;
  padding: 0.3em;
  background: #e3f2fd;
  border-bottom: solid 5px #1565c0;
  margin: 0 0 15px;
}

.c2 {
  font-size: 1.2em;
  font-weight: 600;
  margin: 0 0 15px;
  color: #000;
}

.c3 {
  font-size: 1.2em;
  padding: 0.3em;
  border: none;
  border-bottom: 1px solid #9e9e9e;
  width: 1.8em;
  text-align: center;
}

.c4 {
  font-size: 1em;
  font-weight: 600;
  color: #1565c0;
  padding: 0.3em 0.8em;
  background: #fafafa;
  border: 1px solid #1565c0;
  border-radius: 0.1em;
}

.c4:hover,
.c4:active,
.c4:focus {
  color: #fafafa;
  background-color: #1565c0;
}

.c1 {
  padding: 10px 15px;
}

<div>
  <h1
    className="c0"
  >
    ファイターズ選手名鑑2019
  </h1>
  <div
    className="c1"
  >
    <p
      className="c2"
    >
      背番号が
      <input
        className="c3"
        onChange={[Function]}
        value=""
      />
      の選手は誰?
    </p>
    <button
      className="c4"
      onClick={[Function]}
    >
      送信
    </button>
  </div>
</div>
`;
